sklearn/impute/tests/test_impute.py

```diff
@@ -1383,3 +1383,64 @@ def test_imputation_order(order, idx_order):
                                random_state=0).fit(X)
         idx = [x.feat_idx for x in trs.imputation_sequence_]
         assert idx == idx_order
+
+
+@pytest.mark.parametrize(
+    "missing_value",
+    [-1, np.nan]
+)
+def test_simple_imputation_inverse_transform(missing_value):
+    # Test inverse_transform feature for np.nan
+    X_1 = np.array([
+        [9, missing_value, 3, -1],
+        [4, -1, 5, 4],
+        [6, 7, missing_value, -1],
+        [8, 9, 0, missing_value]
+    ])
+
+    X_2 = np.array([
+        [5, 4, 2, 1],
+        [2, 1, missing_value, 3],
+        [9, missing_value, 7, 1],
+        [6, 4, 2, missing_value]
+    ])
+
+    X_3 = np.array([
+        [missing_value, missing_value, missing_value, missing_value],
+        [missing_value, missing_value, missing_value, missing_value],
+        [missing_value, missing_value, missing_value, missing_value],
+        [missing_value, missing_value, missing_value, missing_value]
+    ])
+
+    X_4 = np.array([
+        [missing_value, missing_value, 2, missing_value, missing_value, 5],
+        [missing_value, missing_value, 3, missing_value, missing_value, 6],
+        [missing_value, missing_value, 4, missing_value, missing_value, 7],
+        [missing_value, missing_value, 5, missing_value, missing_value, 8]
+    ])
+
+    imputer = SimpleImputer(missing_values=missing_value, strategy='mean',
+                            add_indicator=True)
+
+    X_1_trans = imputer.fit_transform(X_1)
```

Comment:
Iterate over X_1 ... X_4 in a for loop

Reply:
The test case for X_2 relies on X_1, so kept them separate. Added a loop over X_3 and X_4.

Reply:
I don't see the difference. You create an imputer, transform, inverse_transform, and check the original X with the twice transformed array no?

Reply:
I agree with @glemaitre that leaving these two out of the loop looks inexplicable to the reader.

Reply:
X_2 is different in the sense that we only perform `transform` on it using the imputer that is fit on X_1. For X_3 and X_4, we do a fresh `fit_transform`.